go/pkg/coordinator/grpc/segment_service.go

```diff
@@ -92,7 +92,14 @@ func (s *Server) DeleteSegment(ctx context.Context, req *coordinatorpb.DeleteSeg
 		res.Status = failResponseWithError(common.ErrSegmentIDFormat, errorCode)
 		return res, nil
 	}
-	err = s.coordinator.DeleteSegment(ctx, parsedSegmentID)
+	collectionID := req.GetCollection()
+	parsedCollectionID, err := types.Parse(collectionID)
+	if err != nil {
+		log.Error(err.Error(), zap.String("collection.id", collectionID))
```

Comment:
is it possible CollectionId is np or badly formed and zap.string() errors?

Reply:
`collectionID` here is a string, so I don't think so
(copied from existing parsing logic a few lines above)